base/path.jl

```diff
@@ -594,3 +595,22 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+"""
+    username() -> String
+
+    Return the “login name” of the user.
+    This function checks the environment variables LOGNAME, USER, LNAME and USERNAME, in order, and
+    returns the value of the first one which is set to a non-empty string.
+    If none are set, the login name from the password database is returned
+"""
+
+function username()
+    for varname in ("LOGNAME", "USER", "LNAME",  "USERNAME")
```

Comment:
I think we need to think carefully about these environment variables.  These are copied from Python, but most other language seem to use just `getpwuid` or similar.  Do we want them at all?  If so, do we want all of them?

Reply:
For example, if `username()` only returned the `getpwuid` password, someone could easily still do `get(ENV, "USER", username())` in a context where they wanted the username to be overridable.